### PR TITLE
Fixed an issue that caused an error when the theme list is emmpty

### PR DESCRIPTION
**Pull request by rcrdortiz**
Related to:
- https://github.com/Automattic/dotcom-forge/issues/4571

## Proposed Changes

* Guarded against an empty theme list when the event is triggered

## Testing Instructions

* Open the Calypso live link
* Go to the LiTS
* Filter the LiTS so that you only have one theme on the results list.
* Click that theme.
* There should be no console errors.
* The t.gif event `calypso_themeshowcase_theme_click` should contain an empty theme list.

No before and after images are provided since this is mainly a BE task.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/themes/events/theme-showcase-tracks.ts [hunk 1]
```diff
@@ -51,7 +51,7 @@ const getThemeShowcaseEventRecorder = (
 				is_active_theme: isActiveTheme( themeId ),
 				style_variation: styleVariation,
 				results_rank: adjustedPosition,
-				results: themes.map( property( 'id' ) ).join(),
+				results: themes?.map( property( 'id' ) ).join() ?? [],
```
Comment (dsas) [inline client/my-sites/themes/events/theme-showcase-tracks.ts]: Non-blocking, i'm ok either way: Is this a good opportunity to remove lodash's `property` ?

Comment (dsas) [inline client/my-sites/themes/events/theme-showcase-tracks.ts]: Given this is typescript isn't `themes` always an array, and hence map doesn't need to be optionally chained?

**Review by dsas — COMMENTED** [review]
I can't reproduce the initial problem. Here are my steps

1. Searched for `eventual` I get one of our themes and two community themes
2. Changed the tier to `free` which removes the community themes
3. Click the Eventual theme

There's no error in the console log, the `results` value in the tracks event is `results: eventual`

File: client/my-sites/themes/events/theme-showcase-tracks.ts (hunk 1 above)
Comment (rcrdortiz) [reply client/my-sites/themes/events/theme-showcase-tracks.ts]: The problem with TypeScript is that it just works on build, during runtime you really can get an empty or null value. TypeScript would just avoid that when you use the type you always supply it.
[Here](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/themes/selectors/get-themes-for-query-ignoring-page.js#L31) you can see that themes can be null at runtime.

The problem we're having is that we did a null value at this point in production for some reason.
This is the error we're dealing with: Cannot read properties of null (reading 'map')

Comment (rcrdortiz) [reply client/my-sites/themes/events/theme-showcase-tracks.ts]: Sure

**Comment by rcrdortiz** [comment]
> I can't reproduce the initial problem. Here are my steps
> 
> 1. Searched for `eventual` I get one of our themes and two community themes
> 2. Changed the tier to `free` which removes the community themes
> 3. Click the Eventual theme
> 
> There's no error in the console log, the `results` value in the tracks event is `results: eventual`

I couldn't repro either, I haven't found a way (without setting it manually) to make the themes list not an array. It's one of those situations that should never happen, but if you look at the code, in theory, it could happen (there are a few nulls returned).

File: client/my-sites/themes/events/theme-showcase-tracks.ts (hunk 1 above)
Comment (rcrdortiz) [reply client/my-sites/themes/events/theme-showcase-tracks.ts]: Done! thanks!

**Review by dsas — APPROVED** [review]
As I said previously, I can't repro the issue but this change makes sense, I get no errors and everything works as it should.

**Comment by dsas** [comment]
The same tests failed twice in a row 😞 might need to rebase or go looking

**Comment by rcrdortiz** [comment]
> The same tests failed twice in a row 😞 might need to rebase or go looking

Rebasing did the trick.

**Pull request merged**